### PR TITLE
Have TensorMechanics Action select the correct kernels for the coordinate system automatically

### LINKED ISSUE #7555: Make TensorMechanicsAction select the correct stress divergence kernels based on the current coordinate system
### Description of the enhancement or error report
Currently the `TensorMechanicsAction` only sets up cartesian kernels. It would be trivial to query the coordinate system and set up the appropriate kernels

### Rationale for the enhancement or information for reproducing the error
Will make using TM even easier.

### Identified impact
None, just added capability... unless somebody has been using cartesian kernels in spherical/cylindrical coordinates :-) then we fix their inputs with this change.

Tag @sapitts

### PR DESCRIPTION
**Pull request by dschwen**
This gets rid of the coordinate system specific actions and thus removes a potential error source and simplifies the syntax.

Closes #7555

### CODE REVIEW

**Comment by dschwen** [comment]
@sapitts and @sveerara, comments welcome (this touches the `DynamicTensorMechanicsAction` as well)

File: modules/tensor_mechanics/src/actions/TensorMechanicsAxisymmetricRZAction.C [hunk 1]
```diff
@@ -6,74 +6,14 @@
...
-    params.set<unsigned int>("component") = i;
-    params.set<NonlinearVariableName>("variable") = displacements[i];
-  if (isParamValid("save_in"))
-    params.set<std::vector<AuxVariableName> >("save_in") = {save_in[i]};
-  if (isParamValid("diag_save_in"))
-    params.set<std::vector<AuxVariableName> >("diag_save_in") = {diag_save_in[i]};
-
-    _problem->addKernel("StressDivergenceRZTensors", kernel_name, params);
-  }
+  mooseDeprecated("Use the 'TensorMechanics' action instead. It autodetects the coordiante system.");
```
Comment (sapitts) [inline modules/tensor_mechanics/src/actions/TensorMechanicsAxisymmetricRZAction.C]: coordinate

**Comment by sapitts** [comment]
The mooseframework wiki tutorial @tonkmr wrote ([part 2: Axisymmetric Simulation](http://mooseframework.org/wiki/PhysicsModules/TensorMechanics/BasicTutorial/basics_part_2/)) will also need to be updated when this merge goes through.

Thanks for making this change!

File: modules/tensor_mechanics/src/actions/TensorMechanicsAxisymmetricRZAction.C (hunk 1 above)
Comment (dschwen) [inline modules/tensor_mechanics/src/actions/TensorMechanicsAxisymmetricRZAction.C]: Ha, good catch. Grepping for that typo I discovered it is also here:
https://github.com/libMesh/libmesh/blob/master/contrib/metis/GKlib/pdb.c#L442

**Comment by dschwen** [comment]
The linked tutorial has more issues. Further down it references the `TensorMechanics` subblock in `Kernels` (which is currently not what that block is named). It also introduced the `use_displaced_mesh` up top without explaining what it does (done further down).

Once this is merged we can simply erase the paragraph about changing the Kernels action and everything will be consistent.

**Comment by sveerara** [comment]
Thanks for making this change Daniel. I will edit the dynamics wiki page once this change is merged.

**Comment by permcody** [comment]
Looks good! I'm merging this @sapitts and @sveerara.

**Comment by dschwen** [comment]
I made the on wiki changes. @sapitts is there any reason the strain calculator should not just autodetect the coordinate system as well?

**Comment by sapitts** [comment]
@dschwen it would make a great deal of sense to have the strain calculator also autodectect the coordinate system.  Because we have three different versions of the strain calculator for each coordinate system (linearized-small, linearized-incremental, and finite), an enum in the action to determine the correct strain formulation.  @bwspenc and I talked with @friedmud about a nested modules action with some similar functionality, e.g.
```
[Modules]
  [./TensorMechanics]
    deformation_theory = small | linearized_incremental | finite
    [./elasticity tensor]
      ..stuff
    [../]
  [../]
[]
```
where the combination of the coordinate system and the deformation_theory flag could be used to properly set both the kernel and the strain calculator in the action.  One could almost set the stress as well, except in the cases for plasticity, ect, that require a special stress calculator.  What do you think about the nested 'overlord' action?

Thanks for updating the wiki!

**Comment by dschwen** [comment]
Yeah, I was there for that discussion. This action would make sense for the deformation theory. 

The issue about the coordinate system is unrelated, though. We already have the coordinate system type available as global data. In the current framework we could make the strain calculator do the right thing depending on the coordinate system. There is no reason to make the user specify the coordinate system twice.

**Pull request merged**